applications/FluidDynamicsApplication/custom_utilities/cfd_utilities.cpp

```diff
@@ -0,0 +1,429 @@
+    v1[1] = pGeometry[1].Y() - pGeometry[0].Y();
+    v1[2] = pGeometry[1].Z() - pGeometry[0].Z();
+
+    v2[0] = pGeometry[2].X() - pGeometry[0].X();
+    v2[1] = pGeometry[2].Y() - pGeometry[0].Y();
+    v2[2] = pGeometry[2].Z() - pGeometry[0].Z();
+
+    MathUtils<double>::CrossProduct(rNormal, v1, v2);
+    rNormal *= 0.5;
+}
+
+double CalculateConditionWallHeight(const ConditionType& rCondition,
+                                    const array_1d<double, 3>& rNormal)
+{
+    KRATOS_TRY
+
+    array_1d<double, 3> normal = rNormal / norm_2(rNormal);
+
+    const ElementType& r_parent_element = rCondition.GetValue(NEIGHBOUR_ELEMENTS)[0];
+
+    const GeometryType& r_parent_geometry = r_parent_element.GetGeometry();
+    const GeometryType& r_condition_geometry = rCondition.GetGeometry();
+
+    auto calculate_cell_center = [](const GeometryType& rGeometry) -> array_1d<double, 3> {
+        const int number_of_nodes = rGeometry.PointsNumber();
+        array_1d<double, 3> cell_center = ZeroVector(3);
+        for (int i_node = 0; i_node < number_of_nodes; ++i_node)
+        {
+            noalias(cell_center) =
+                cell_center + rGeometry[i_node].Coordinates() *
+                                  (1.0 / static_cast<double>(number_of_nodes));
+        }
+
+        return cell_center;
+    };
+
+    const array_1d<double, 3>& parent_center = calculate_cell_center(r_parent_geometry);
+
+    const array_1d<double, 3>& condition_center =
+        calculate_cell_center(r_condition_geometry);
+
+    return inner_prod(condition_center - parent_center, normal);
+
+    KRATOS_CATCH("");
+}
+
+void CalculateNumberOfNeighbourConditions(ModelPart& rModelPart)
+{
+    KRATOS_TRY
+
+    ModelPart::NodesContainerType& r_nodes = rModelPart.Nodes();
+
+    // reseting all nodal variables which needs to be calculated
+    VariableUtils().SetNonHistoricalVariableToZero(NUMBER_OF_NEIGHBOUR_CONDITIONS, r_nodes);
+
+    // updating
+    block_for_each(rModelPart.Conditions(), [](ConditionType& r_cond) {
+        ConditionType::GeometryType& r_geometry = r_cond.GetGeometry();
+        for (IndexType i_node = 0; i_node < r_geometry.PointsNumber(); ++i_node)
+        {
+            NodeType& r_node = r_geometry[i_node];
+            r_node.SetLock();
+            r_node.GetValue(NUMBER_OF_NEIGHBOUR_CONDITIONS) += 1;
+            r_node.UnSetLock();
+        }
+    });
+
+    rModelPart.GetCommunicator().AssembleNonHistoricalData(NUMBER_OF_NEIGHBOUR_CONDITIONS);
+
+    KRATOS_CATCH("");
+}
+
+double CalculateReactionBasedYPlusUTau(array_1d<double, 3>& rFrictionVelocity,
+                                       const array_1d<double, 3>& rReaction,
+                                       const array_1d<double, 3>& rNormal,
+                                       const double Density,
+                                       const double KinematicViscosity,
+                                       const double WallHeight)
```

Comment:
```suggestion
double CalculateReactionBasedYPlusUTau(
    array_1d<double, 3>& rFrictionVelocity,
    const array_1d<double, 3>& rReaction,
    const array_1d<double, 3>& rNormal,
    const double Density,
    const double KinematicViscosity,
     const double WallHeight)
```
Please do so in the header and the rest of the methods.

Reply:
done :)